master/cmd/stream-gen/main.go

```diff
@@ -238,6 +240,100 @@ func (b *Builder) String() string {
 	return b.builder.String()
 }
 
+func genTypescript(streamables []Streamable) ([]byte, error) {
+	b := Builder{}
+	typeAnno := func(f Field) ([2]string, error) {
+		x := map[string]([2]string){
+			"JSONB":            {"any", "{}"},
+			"string":           {"string", ""},
+			"bool":             {"bool", "false"},
+			"int":              {"number", "0"},
+			"int64":            {"number", "0"},
+			"[]int":            {"Array<number>", "[]"},
+			"time.Time":        {"string", ""},
+			"*time.Time":       {"string | undefined", "undefined"},
+			"model.TaskID":     {"string", ""},
+			"model.RequestID":  {"number", "0"},
+			"*model.RequestID": {"number | undefined", "undefined"},
+			"model.State":      {"string", ""},
+		}
+		out, ok := x[f.Type]
+		if !ok {
+			return [2]string{"", ""}, fmt.Errorf("no type annotation matches %q", f.Type)
+		}
+		return out, nil
+	}
+	b.Writef("// Code generated by stream-gen. DO NOT EDIT.\n")
+	b.Writef("\n")
+	b.Writef("import { isEqual } from 'lodash';\n")
+	b.Writef("\n")
+	b.Writef("import { Streamable, StreamSpec } from '.';\n")
+	b.Writef("\n")
+	for _, s := range streamables {
+
+		source := s.Args["source"]
+		entity := strings.ToLower(strings.TrimSuffix(s.Name, "SubscriptionSpec"))
+		caser := cases.Title(language.English)
```

Comment:
I think it's safer to just generate `StorageBackendSpec` because it's possible to have `StorageBackendSubscriptionSpec` and `StorageSubscriptionSpec` from backend and we don't want to risk duplicating class name when generating

Reply:
Not a blocker: when `s.name = "StorageBackendSubscriptionSpec"`, `case.String(entity) = "Storagebackend"`.

Reply:
I think I misunderstood your first comment, I get it now. More than one word can be tricky, thank you for pointing it out